### ISSUE TITLE
Replace tables for more responsive design

### ISSUE THREAD

**Issue by audreytoskin** [lojban/cll#245]
(NOTE: Consistent font rendering will be important for this issue, so we may want to resolve Issue #244 first.)

Issues like #163 "Ugly Wrapping" may continue to come up as development continues on a large document like the CLL. I would argue that for many of the cases where text wrapping is or could become an issue, a table is simply the wrong type of markup in the first place. Especially for all the interlinear glosses of the example sentences. A table is not the most responsive, semantic, or accesible way to markup an interlinear gloss.

I've spent the past several days considering alternative solutions, and while I have my opinion as to which is best, I wanted to you see the options available and decide for yourself. And since we'll be talking so much about the code for producing the documents, I thought it would help to see a demo of all possible solutions, so I've put together an online demo. I picked the beta of the new version of Mozilla Thimble because it lets you upload arbitrary files as well as lets you see the markup and rendered page side-by-side. See the mini-project [on Thimble](https://thimble.mozilla.org/anonymous/7574171f-f2bd-4d7e-a26e-d560ec144beb/30247).

This new version of Thimble _is_ still beta, though, and it sometimes take a little while to load. If you have any trouble, let me know, and I can upload the demo somewhere else.

**Comment by lagleki** [comment]
I'm afraid I can't see other solutions there. clicking on "Pair blocks"
etc. leads on non  existent pages.

I'd suggest (if that's possible) to
1. try different solutions on the REAL book (this CLL) and see which one
definitely not work.
2. try making an automatic converter from the current table format to
different formats and producing pdfs from them.

Also if we get working solutions without tables the only requirement left
would be "what new they can bring us" so that we change from table format
to another one.

2016-02-04 11:30 GMT+03:00 Andrew Toskin notifications@github.com:

**Comment by audreytoskin** [comment]
> clicking on "Pair blocks" etc. leads on non existent pages.

Yeah, sorry for the confusion: If you see the demo on the [rendered output page](https://d157rqmxrxj6ey.cloudfront.net/terrycloth/30247/#other-solutions) you can click the links like normal. [In the Thimble editor](https://thimble.mozilla.org/projects/30247/remix), though, apparently clicking on links doesn't work; you have to click on the files in the file list, on the left side of the screen.

If that still doesn't work, you can see the demo [on my personal website](https://andrew.tosk.in/demo/interlinear-gloss/#other-solutions), or [download a zip archive](https://andrew.tosk.in/demo/interlinear-gloss.zip) of all the files.

> try different solutions on the REAL book (this CLL) and see which one definitely not work.

I'm actually already using real examples taken from the CLL book. I only quote parts of the CLL, and it's all in web format, but I think this demo should be good enough for us to decide which solution we want to use. Then we can figure out the most efficient way to convert the markup.

As far as reformatting the interlinear glosses, the only difference between the web and print versions I can think of should be whether DocBook/Prince works with JavaScript. (And I _think_ they do allow JavaScript. Right?)

**Comment by lagleki** [comment]
2016-02-05 2:24 GMT+03:00 Andrew Toskin notifications@github.com:

> As far as reformatting the interlinear glosses, the only difference
> between the web and print versions I can think of should be whether
> DocBook/Prince works with JavaScript. (And I _think_ they do allow
> JavaScript. Right?)

Question to Robin.
And I meant that the whole book needs to presented in the final formats
with visual explanations of how future modifications of CLL can break
tables while your solutions would still support nice formatting.

**Comment by audreytoskin** [comment]
Okay, the use of JavaScript can be a question for @rlpowell 

Reformatting the entire book would be a non-trivial effort, especially if I were to do it _three time_ just to demonstrate the three possible solutions which I proposed. So no, sorry, I'm not going to do that. My demo pages already take Lojban example sentences from the book, gives a high level overview of how the solutions would work, reformats the given example sentences so you can see what the new markup looks like, and then weighs some of the benefits and trade-offs. If you're still unsure after reading all that, I'll be happy to answer questions, or maybe even create a few more examples for each of the possible solutions. But I'm not going to reformat the entire book three times.

**Comment by lagleki** [comment]
2016-02-06 3:54 GMT+03:00 Andrew Toskin notifications@github.com:

> Okay, the use of JavaScript can be a question for @rlpowell
> https://github.com/rlpowell
> 
> Reformatting the entire book would be a non-trivial effort, especially if
> I were to do it _three time_ just to demonstrate the three possible
> solutions which I proposed. So no, sorry, I'm not going to do that. My demo
> pages already take Lojban example sentences from the book, gives a high
> level overview of how the solutions would work, reformats the given example
> sentences so you can see what the new markup looks like, and then weighs
> some of the benefits and trade-offs. If after reading all that, you're
> still unsure about something, I'm happy to answer questions, or maybe even
> create a few more examples for each of the possible solutions. But I'm not
> going to reformat the entire book three times.

If you are absolutely sure one way works you could do it one time.
